polygeist/lib/Dialect/Polygeist/Transforms/RaiseToAffine.cpp

```diff
@@ -69,128 +56,142 @@ struct ForOpRaising : public OpRewritePattern<scf::ForOp> {
-        auto *affineYieldOp = newBlock.getTerminator();
-        rewriter.eraseOp(affineYieldOp);
-      }
+    ConstantIndexOp cstOp = loop.getStep().getDefiningOp<ConstantIndexOp>();
+    AffineForOp affineLoop = rewriter.create<AffineForOp>(
+        loop.getLoc(), lbs, lbMap, ubs, ubMap, cstOp ? cstOp.value() : 1,
+        loop.getIterOperands());
 
-      SmallVector<Value> vals;
-      rewriter.setInsertionPointToStart(&affineLoop.getRegion().front());
-      for (Value arg : affineLoop.getRegion().front().getArguments()) {
-        if (rewrittenStep && arg == affineLoop.getInductionVar()) {
-          arg = rewriter.create<AddIOp>(
-              loop.getLoc(), loop.getLowerBound(),
-              rewriter.create<MulIOp>(loop.getLoc(), arg, loop.getStep()));
-        }
-        vals.push_back(arg);
-      }
-      assert(vals.size() == loop.getRegion().front().getNumArguments());
-      rewriter.mergeBlocks(&loop.getRegion().front(),
-                           &affineLoop.getRegion().front(), vals);
+    auto mergedYieldOp =
+        cast<scf::YieldOp>(loop.getRegion().front().getTerminator());
 
-      rewriter.setInsertionPoint(mergedYieldOp);
-      rewriter.create<AffineYieldOp>(mergedYieldOp.getLoc(),
-                                     mergedYieldOp.getOperands());
-      rewriter.eraseOp(mergedYieldOp);
+    Block &newBlock = affineLoop.getRegion().front();
 
-      rewriter.replaceOp(loop, affineLoop.getResults());
+    // The terminator is added if the iterator args are not provided.
+    // see the ::build method.
+    if (affineLoop.getNumIterOperands() == 0) {
+      Operation *affineYieldOp = newBlock.getTerminator();
+      rewriter.eraseOp(affineYieldOp);
+    }
 
-      return success();
+    SmallVector<Value> vals;
+    rewriter.setInsertionPointToStart(&affineLoop.getRegion().front());
+    for (Value arg : affineLoop.getRegion().front().getArguments()) {
+      if (rewrittenStep && arg == affineLoop.getInductionVar())
+        arg = rewriter.create<AddIOp>(
+            loop.getLoc(), loop.getLowerBound(),
+            rewriter.create<MulIOp>(loop.getLoc(), arg, loop.getStep()));
+      vals.push_back(arg);
     }
-    return failure();
+    assert(vals.size() == loop.getRegion().front().getNumArguments());
+    rewriter.mergeBlocks(&loop.getRegion().front(),
+                         &affineLoop.getRegion().front(), vals);
+
+    rewriter.setInsertionPoint(mergedYieldOp);
+    rewriter.create<AffineYieldOp>(mergedYieldOp.getLoc(),
+                                   mergedYieldOp.getOperands());
+    rewriter.eraseOp(mergedYieldOp);
+
+    [[maybe_unused]] auto getParentFunction = [](LoopLikeOpInterface loop) {
+      Operation *parentOp = loop;
+      do {
+        parentOp = parentOp->getParentOp();
+      } while (parentOp && !isa<func::FuncOp>(parentOp));
+      assert(parentOp && "Failed to find parent function");
+      return cast<func::FuncOp>(parentOp);
+    };
+
+    DEBUG_WITH_TYPE(REPORT_DEBUG_TYPE, {
```

Comment:
Why not `LLVM_DEBUG`?

Reply:
Because we already use LLVM_DEBUG to print traces in this file (we can only use one DEBUG_TYPE). 
Also this is the convention to emit more verbose debug traces in files that already have DEBUG_TYPE defined (LoopVectorize.cpp in llvm is an example).